Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -433,26 +434,29 @@ WarpXOpenPMDPlot::WriteOpenPMDParticles (const amrex::Vector<ParticleDiag>& part
 #ifdef WARPX_DIM_RZ
     real_names.push_back("theta");
 #endif
-    if(pc->DoFieldIonization()){
-       int_names.push_back("ionizationLevel");
-       // int_flags specifies, for each integer attribs, whether it is
-       // dumped as particle record in a plotfile. So far, ionization_level is the only
-       // integer attribs, and it is automatically dumped as particle record
-       // when ionization is on.
-       int_flags.resize(1, 1);
-       tmp.AddIntComp(false);
-    }
 
-#ifdef WARPX_QED
-    if( pc->has_breit_wheeler() ) {
-        real_names.push_back("opticalDepthBW");
-        tmp.AddRealComp(false);
-    }
-    if( pc->has_quantum_sync() ) {
-        real_names.push_back("opticalDepthQSR");
-        tmp.AddRealComp(false);
-    }
-#endif
+    // add runtime real comps to tmp
+    for (int ic = 0; ic < pc->NumRuntimeRealComps(); ++ic) { tmp.AddRealComp(false); }
+
+    // get the names of the real comps
+    real_names.resize(pc->NumRealComps());
+    auto runtime_rnames = pc->getParticleRuntimeComps();
```

Comment:
Here we need to be careful with `_` in names, as we use them currently as separators for vector/tensor components.

In the current form this re-introduces #1777

Reply:
I have fixed this by converting these names to camelCase, as they are in `development`. 

We need a more systematic way of handling scalar versus vector component names, but for now I think we should merge this as it is blocking a few different PRs.

Reply:
Clarified:
> converting these names

is only the runtime added attributes, so momentum et al (added above) that are added by default are unaffected.